File: algorithms-Python/eratosthenes_sieve.py
n = raw_input("PLease enter an integer that you'd like to find prime numbers equal to or smaller than: ")
n = n.strip()
n = int(n)

sieve = [1]*(n+1)

k = 2

while k*k <= n:
    i = k
    while i <= n/k:
        sieve[k*i] = 0
        i += 1

    while True:
        k += 1
        if sieve[k] != 0:
            break

i = 2
while i <= n:
    if sieve[i] == 1:
        print i
    i += 1

File: algorithms-R/eratosthenes-sieve.R
## eratosthenes-sieve.R
##
## Find prime numbers by Eratosthenes's sieve
##
## 2015-01-02 YY
## 2016-11-07 YY

sieve <- function(x) {
  ## input: a positive number x
  ## return: vector of prime numbers equal to or smaller than x
  x <- floor(x)
  indicators <- rep(1, x)
  indicators[1] <- 0
  k <- 2
  while (k^2 <= x) {
    i = k
    while (i <= x/k) {
      indicators[k*i] <- 0
      i = i + 1
    }   
    while (TRUE) {
      k = k + 1
      if (indicators[k] != 0) break
    }
  }
 prime <- 1:x
 prime <- prime[as.logical(indicators)]
 return(prime)
}
